Makefile

```diff
@@ -214,20 +220,31 @@ docker-push-agent:
 docker-push-router:
 	docker push ${KO_DOCKER_REPO}/${ROUTER_IMG}
 
-docker-build-sklearn:
-	cd python && docker buildx build --build-arg BASE_IMAGE=${BASE_IMG} -t ${KO_DOCKER_REPO}/${SKLEARN_IMG} -f sklearn.Dockerfile .
+docker-create-registry:
+	docker run -d -p ${REGISTRY_PORT}:5000 --restart always --name ${REGISTRY_NAME} registry:2
+
+docker-rm-registry:
+	docker rm -f ${REGISTRY_NAME}
+
+docker-build-base-images: docker-create-registry
+	cd python && docker buildx build --build-arg BASE_IMAGE=${PYTHON_BASE_IMG} -t ${BUILD_BASE_IMG} -f build_base_image.Dockerfile --push .
+	cd python && docker buildx build --build-arg BASE_IMAGE=${PYTHON_BASE_IMG} -t ${PROD_BASE_IMG} -f prod_base_image.Dockerfile --push .
+
+docker-build-sklearn: docker-build-base-images
```

Comment:
@yuzisun / @ckadner In this PR as @ckadner suggested I have separated out the common layers from the runtime images and created two base images. One is for build stage and another one is for prod stage. Using locally built base images does not pose any problems for building runtime images as long as we use `docker` driver. However `docker-container` is the only driver that supports multi arch builds. We are using this driver in github workflow for publishing images. But If we use the `docker-container` driver it can't access those local base images as it runs on a separate container unlike the former which runs locally. It can only access the images form a registry. I have tackled this issue in workflow by pushing the base images to the local registry and pulling from it. However for local development this makes things complicated. Should we warn the users not to use the `docker-container` driver or setup a local registry in the local machine as I provided in this `Makefile`. But the latter option has some downsides too. 

References:
https://docs.docker.com/build/drivers/
https://docs.docker.com/build/drivers/docker-container/
https://docs.docker.com/build/drivers/docker/
https://github.com/moby/buildkit/issues/2343#issuecomment-930094423